core/deployment/src/main/java/io/quarkus/deployment/builditem/ConnectionFactoryWrapperBuildItem.java

```diff
@@ -0,0 +1,20 @@
+package io.quarkus.deployment.builditem;
+
+import java.util.function.Function;
+
+import io.quarkus.builder.item.SimpleBuildItem;
+
+/**
+ * A build item that can be used to wrap ConnectionFactory
```

Comment:
It's not clear what `ConnectionFactory` is being wrapped. It should be JMS-specific, or?

Reply:
-1 

I wonder if we should introduce a JMS SPI module instead, e.g. something like `quarkus-jms-spi`. This module can be consumed by all JMS extensions.

Reply:
You mean add a module `jms-spi` in `core`?

Reply:
It depends. It could be a module in the `quarkus/core` or in `quarkiverse/quarkus-artemis`. @maxandersen WDYT?

Reply:
Since this is a common module for all JMS extension, I prefer to add it in `quarkus/core`.

Reply:
@gsmet Do you think it's ok to place this spi module in the core?

Reply:
Agree that it should at least be in extensions. I think a common quarkiverse repo makes the most sense but its so small that including it is not unreasonable. The one thing that gives me pause about this is just that people have been complaining about IDE setups being slow to index, so adding another major API jar might make things worse. In the long term we should move to some internal split layout to address that problem, but thats not happening in the short term.

Reply:
Thanks all for reviewing. So it looks like we agree to add this BuildItem in a spi extension. Something similar with https://github.com/quarkusio/quarkus/tree/main/extensions/jaxrs-spi, is it OK?

Reply:
@gsmet @n1hility can you take a review and is it OK for merging?

Reply:
> So it looks like we agree to add this BuildItem in a spi extension

Not sure what makes you think that. I think all of us are not overly excited to have this in core and would prefer a Quarkiverse extension. I will merge it anyway but don't be surprised if at some point we decide to move it out. Core is extremely crowded and the bar to add something in it is very high.

I will do a last review round.

Reply:
OK, I see and thankd for reviewing.